dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/RouterChain.java

```diff
@@ -271,14 +272,23 @@ public void loop(boolean notify) {
         if (firstBuildCache.compareAndSet(true,false)) {
             buildCache(notify);
         }
-        if (notify) {
-            if (loopPermitNotify.tryAcquire()) {
-                loopPool.submit(new NotifyLoopRunnable(true, loopPermitNotify));
+
+        try {
+            if (notify) {
```

Comment:
Nested `if`s make the code unreadable. Here would be better to merge the nested `if`s into a single `if`.
The same applies to the `if` in the `else` statement.

I would suggest:
```
if( notify && loopPermitNotify.tryAcquire() ) {
  loopPool.submit(new NotifyLoopRunnable(true, loopPermitNotify));
}else if(loopPermit.tryAcquire()) {
  loopPool.submit(new NotifyLoopRunnable(false, loopPermit));
}
```

Reply:
The above code logic is not exactly the same as the original one:
if notify==true and loopPermitNotify.tryAcquire() == false,  **no attempt to acquire the loopPermit lock, and no operation will be performed.**

Reply:
My suggestion:
```
if( notify && loopPermitNotify.tryAcquire() ) {
  loopPool.submit(new NotifyLoopRunnable(true, loopPermitNotify));
}else if(loopPermit.tryAcquire()) {
  loopPool.submit(new NotifyLoopRunnable(false, loopPermit));
}
```

Current code:
```
if (notify) {
                if (loopPermitNotify.tryAcquire()) {
                    loopPool.submit(new NotifyLoopRunnable(true, loopPermitNotify));
                }
            } else {
                if (loopPermit.tryAcquire()) {
                    loopPool.submit(new NotifyLoopRunnable(false, loopPermit));
                }
            }
```

Semantically they are the same. See the following table with all possible outcomes of that code.

| notify | loopPermitNotify.tryAcquire() | loopPermit.tryAcquire() | Current Code                           | My Suggestion                              | is same |
|--------|-------------------------------|-------------------------|-----------------------------------------------|--------------------------------------------|---------|
| true   | true                          | true                    | NotifyLoopRunnable ( true , loopPermitNotify) | NotifyLoopRunnable(true, loopPermitNotify) | ✅       |
| false  | true                          | true                    | NotifyLoopRunnable ( false , loopPermit)      | NotifyLoopRunnable(false, loopPermit)      | ✅       |
| true   | false                         | true                    | NotifyLoopRunnable ( false , loopPermit)      | NotifyLoopRunnable(false, loopPermit)      | ✅       |
| false  | false                         | true                    | NotifyLoopRunnable ( false , loopPermit)      | NotifyLoopRunnable(false, loopPermit)      | ✅       |
| true   | true                          | false                   | NotifyLoopRunnable ( true , loopPermitNotify) | NotifyLoopRunnable(true, loopPermitNotify) | ✅       |
| false  | true                          | false                   | NO ACTION                                     | NO ACTION                                  | ✅       |
| true   | false                         | false                   | NO ACTION                                     | NO ACTION                                  | ✅       |
| false  | false                         | false                   | NO ACTION                                     | NO ACTION                                  | ✅       |